src/Types.ts

```diff
@@ -241,64 +241,64 @@ export type IntentArgs = {
   expiryBlock?: BigIntish
```

Comment:
the DSL args can be named differently so they don't conflict. Also it looks like they're named at the wrong levels:

`IntentArgs` would be:

```
export type DeclarationDefinitionArgs = {
  intents: IntentSegmentArgs[]
  replay?: IntentReplay
  expiryBlock?: BigIntish
}

and `IntentSegmentArgs` would be:

```
export type IntentDefinitionArgs = {
  replay?: IntentReplay
  expiryBlock?: BigIntish
  conditions?: ConditionArgs[]
  actions: ActionArgs[]
}
```